sql_metadata/generalizator.py

```diff
@@ -0,0 +1,106 @@
+"""
+Module used to produce generalized sql out of given query
+"""
+import re
+from typing import List, Optional
+
+import sqlparse
+
+
+class Generalizator:
+    """
+    Class used to produce generalized sql out of given query
+    """
+
+    def __init__(self, sql: Optional[str] = None):
+        self._raw_query = sql
+
+    # SQL queries normalization (#16)
+    @staticmethod
+    def _normalize_likes(sql: str) -> str:
+        """
+        Normalize and wrap LIKE statements
+
+        :type sql str
+        :rtype: str
+        """
+        sql = sql.replace("%", "")
+
+        # LIKE '%bot'
+        sql = re.sub(r"LIKE '[^\']+'", "LIKE X", sql)
+
+        # or all_groups LIKE X or all_groups LIKE X
+        matches = re.finditer(r"(or|and) [^\s]+ LIKE X", sql, flags=re.IGNORECASE)
+        matches = [match.group(0) for match in matches] if matches else None
+
+        if matches:
+            for match in set(matches):
+                sql = re.sub(
+                    r"(\s?" + re.escape(match) + ")+", " " + match + " ...", sql
+                )
+
+        return sql
+
+    @property
+    def comments(self) -> List[str]:
+        """
+        Gets comments from SQL query
+
+        :rtype: str
+        """
+        comments = re.findall(r"\s?/\*.+?\*/", self._raw_query)
+        return [x.strip() for x in comments]
+
+    @property
+    def remove_comments(self) -> str:
```

Comment:
I'd rename it to be a noun as it;s a property after all.

So either

```suggestion
    @property
    def without_comments(self) -> str:
```

or

```suggestion
    def remove_comments(self) -> str:
```

Reply:
How do you prefer?

Reply:
Since most/all of them are properties maybe the first one to be in line with most features?

Reply:
ea02fa1